hardhat.config.js

```diff
@@ -0,0 +1,23 @@
+require('@nomiclabs/hardhat-ethers');
+
+const dotenv = require('dotenv');
+
+dotenv.config();
+
+const mainnetFork = {
+  url: `${process.env.JSON_RPC_PROVIDER}`,
+  blockNumber: 14390000,
```

Comment:
wondering why we have blockNumber in the hardhat config? I'd think we'd always want to run the tests against the latest block

Reply:
In the routing-api we make a call to the deployed api to grab the latest block and subtract like 10 and use that:
https://github.com/Uniswap/routing-api/blob/98439fa9697617da7156cac485ae29747291447f/test/integ/quote.test.ts#L129-L144

I think by default hardhat forks from the latest block which we could use, though aren't there benefits from pinning a block (consistent state, caching, etc.)?

Reply:
IMO I think there's benefits to also not pinning a block, being we're constantly testing against different (and more up-to-date) state. Which battletests the code even more (almost like fuzzing) could eventually catch an edge case. I

Should/are we continuing to do it like the routing-api was?

Reply:
I agree - and yeah I think the idea is to do things like routing-api. Let's go with forking from the latest block - 10  which is the default config

Reply:
I think like consistent state is important for a lot of tests, but these integration tests are the final frontier before prod, so I kind of like that the blockchain state can be unpredictable.

Reply:
ok realizing we're keeping this in the routing-api as well, but I think it's worth it to test against recent-ish blockchain state.